tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java

```diff
@@ -139,9 +170,67 @@ public void begin() {
             this.future = SurveyThreadManager.getExecutorService().submit(this);
         }
 
+        final CLDRLocale EN = CLDRLocale.getInstance("en");
+
         @Override
         public Search call() throws Exception {
             // first, simple search of exact values
+            addExactMatches(locale);
+
+            CLDRLocale cLoc = CLDRLocale.getInstance(locale);
+            CLDRLocale parLoc = cLoc.getParent();
+            while (response.isEmpty() && parLoc != null) {
+                addExactMatches(parLoc.getBaseName());
+                parLoc = parLoc.getParent();
+            }
+
+            // add codes (in the locale)
+            addCodes(locale);
+
+            // Add English (if we didn't already search it)
+            if (response.isEmpty() && !cLoc.childOf(EN)) {
```

Comment:
We will probably want control over whether or not to include English, to avoid spurious hits. Eg a check box.

Reply:
But we can start with this.